HW0/P5/P5.txt

```diff
@@ -1 +1,51 @@
 # Your answers here.
+1. 255 seconds
+
+2. Yes. First, split the bags into 8 groups of 32 for each cashier. The 8 cashiers will take 31 seconds to add the total sum of each group. Now, to add the 8 numbers together, it takes 1 second to add 4 pairs of numbers, then 1 second to add the remaining 2 pairs of numbers, and finally 1 second to add the last pair of numbers (i.e. it is a binary tree). Hence, it will take 34 seconds to add up the sum.
+
+3. 8 seconds. log(256)/log(2). Essentially, it is a binary tree.
+
+4. It will take log(N)/log(2)
+
+5. 16 seconds. The binary tree has a depth of 9, hence there are 8 seconds to communicate between levels.  
```

Comment:
A bit more details or a short explanation would be good for questions 3, 4 and 5. I understand your answers as I did the same but if someone hadn't understood the logic, it would be hard to grasp the solutions with your answer.